airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendCheckUpdatesHandler.java

```diff
@@ -0,0 +1,103 @@
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@AllArgsConstructor
+@Slf4j
+public class WebBackendCheckUpdatesHandler {
+
+  private static final int NO_CHANGES_FOUND = 0;
+
+  final ConfigRepository configRepository;
+  final AirbyteGithubStore githubStore;
+
+  public WebBackendCheckUpdatesRead checkUpdates() {
+
+    final int destinationDiffCount = getDestinationDiffCount();
+    final int sourceDiffCount = getSourceDiffCount();
+
+    return new WebBackendCheckUpdatesRead()
+        .destinationDefinitions(destinationDiffCount)
+        .sourceDefinitions(sourceDiffCount);
+  }
+
+  private int getDestinationDiffCount() {
+    final List<Entry<UUID, String>> currentActorDefToDockerImageTag;
+    final Map<UUID, String> newActorDefToDockerImageTag;
+
+    try {
+      currentActorDefToDockerImageTag = configRepository.listStandardDestinationDefinitions(false)
+          .stream()
+          .map(def -> Map.entry(def.getDestinationDefinitionId(), def.getDockerImageTag()))
+          .toList();
+    } catch (final IOException e) {
+      log.error("Failed to get current list of standard destination definitions", e);
+      return NO_CHANGES_FOUND;
+    }
+
+    try {
+      newActorDefToDockerImageTag = githubStore.getLatestDestinations()
+          .stream()
+          .collect(Collectors.toMap(StandardDestinationDefinition::getDestinationDefinitionId, StandardDestinationDefinition::getDockerImageTag));
+    } catch (final InterruptedException e) {
+      log.error("Failed to get latest list of standard destination definitions", e);
+      return NO_CHANGES_FOUND;
+    }
+
+    return getDiffCount(currentActorDefToDockerImageTag, newActorDefToDockerImageTag);
+  }
+
+  private int getSourceDiffCount() {
+    final List<Entry<UUID, String>> currentActorDefToDockerImageTag;
+    final Map<UUID, String> newActorDefToDockerImageTag;
+
+    try {
+      currentActorDefToDockerImageTag = configRepository.listStandardSourceDefinitions(false)
+          .stream()
+          .map(def -> Map.entry(def.getSourceDefinitionId(), def.getDockerImageTag()))
+          .toList();
+    } catch (final IOException e) {
+      log.error("Failed to get current list of standard source definitions", e);
+      return NO_CHANGES_FOUND;
+    }
+
+    try {
+      newActorDefToDockerImageTag = githubStore.getLatestSources()
+          .stream()
+          .collect(Collectors.toMap(StandardSourceDefinition::getSourceDefinitionId, StandardSourceDefinition::getDockerImageTag));
+    } catch (final InterruptedException e) {
+      log.error("Failed to get latest list of standard source definitions", e);
+      return NO_CHANGES_FOUND;
+    }
+
+    return getDiffCount(currentActorDefToDockerImageTag, newActorDefToDockerImageTag);
+  }
+
+  private int getDiffCount(final List<Entry<UUID, String>> initialSet, final Map<UUID, String> newSet) {
+    int diffCount = 0;
```

Comment:
nit:
```
initialSet.stream()
.filter(entry -> {
  final String newDockerImageTag = newSet.get(entry.getKey());
  return newDockerImageTag != null && !entry.getValue().equals(newDockerImageTag))
})
.count();
```

Reply:
as discussed, using count sends us into type issues so we'll stick to old fashioned for loop count here.